posthog/hogql/printer.py

```diff
@@ -1090,6 +1090,12 @@ def visit_lazy_table_type(self, type: ast.LazyJoinType):
     def visit_field_traverser_type(self, type: ast.FieldTraverserType):
         raise ImpossibleASTError("Unexpected ast.FieldTraverserType. This should have been resolved.")
 
+    def visit_unresolved_field_type(self, type: ast.UnresolvedFieldType):
+        if self.dialect == "clickhouse":
+            raise QueryError(f"Unable to resolve field: {type.name}")
+        else:
```

Comment:
Nitpick: Can unwrap else in such cases since it's the main thing that happens. Advantage: main logic on top level not nested. The if is just early exit on some condition.